TELEMETRY.md

```diff
@@ -0,0 +1,43 @@
+# Telemetry in the CodeQL extension for VS Code
+
+If you specifically opt-in to permit GitHub to do so, GitHub will collect usage data and metrics for the purposes of helping the core developers to improve the CodeQL extension for VS Code. This data will not be shared with any parties outside of GitHub and will be retained for a maximum of 30 days.
+
+## Why do you collect data?
+
+GitHub collects usage data and metrics to help us improve CodeQL for VS Code.
+
+## What data is collected
+
+GitHub collects anonymized information related to the usage of the extensions. The data collected are:
+
+- Which commands are run.
+- The time taken for each command.
+- Anonymous GUID to uniquely identify an installation.
+- IP address of the client sending the telemetry data.
+
```

Comment:
~Would it be worth clarifying here that we **do not** collect any information about the nature of the CodeQL queries that are being run?~ You already do this further down. Excellent!